flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/ExecNodeConfiguration.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.planner.plan.nodes.exec;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.configuration.ConfigOption;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.configuration.ReadableConfig;
+import org.apache.flink.table.api.TableConfig;
+import org.apache.flink.table.api.config.ExecutionConfigOptions;
+import org.apache.flink.table.api.config.TableConfigOptions;
+import org.apache.flink.table.planner.delegation.PlannerConfiguration;
+
+import java.time.ZoneId;
+import java.util.Optional;
+
+import static java.time.ZoneId.SHORT_IDS;
+
+/**
+ * Configuration view that combines the {@link PlannerConfiguration} with the {@link
+ * ExecNodeBase#getPersistedConfig()} configuration. The persisted configuration of the {@link
+ * ExecNode} which is deserialized from the JSON plan has precedence over the {@link
+ * PlannerConfiguration}.
+ */
+@Internal
+public class ExecNodeConfiguration implements ReadableConfig {
```

Comment:
Perhaps can you extend from `PlannerConfiguration` directly to reuse the merging logic? TBH I would move `PlannerConfiguration` in `org.apache.flink.configuration` and rename it to something more generic, as it's essentially doing a merge of the configs, choosing one as precedence

Reply:
I agree with this, @twalthr do you also agree? to make PlannerConfiguration more generic, like `MergedConfiguration`?

Reply:
I will leave it as is for now, and make a comment on the relevant issue to maybe do this refactoring when we remove the `TableConfig` from here.